src/main/java/xyz/groundx/caver_ext_kas/wallet/KASWallet.java

```diff
@@ -0,0 +1,353 @@
+        }
+
+        if(!feeDelegatedTransaction.getFeePayer().toLowerCase().equals(feePayerAddress.toLowerCase())) {
+            throw new IllegalArgumentException("Fee payer address are not matched");
+        }
+
+        if(isWeightedMultiSigType(feeDelegatedTransaction, feePayerAddress)) {
+            throw new IllegalArgumentException("Not supported: Using multiple keys in an account is currently not supported.");
+        }
+
+        feeDelegatedTransaction.fillTransaction();
+        String rlp = feeDelegatedTransaction.getRLPEncoding();
+
+        FDUserProcessRLPRequest rlpRequest = new FDUserProcessRLPRequest();
+        rlpRequest.setRlp(rlp);
+        rlpRequest.setFeePayer(feeDelegatedTransaction.getFeePayer());
+        rlpRequest.setSubmit(false);
+
+        if(feeDelegatedTransaction instanceof AbstractFeeDelegatedWithRatioTransaction) {
+            rlpRequest.setFeeRatio(((AbstractFeeDelegatedWithRatioTransaction) feeDelegatedTransaction).getFeeRatioInteger().longValue());
+        }
+
+        FDTransactionResult result = this.walletAPI.requestFDRawTransactionPaidByUser(rlpRequest);
+
+        AbstractFeeDelegatedTransaction tx = (AbstractFeeDelegatedTransaction)TransactionDecoder.decode(result.getRlp());
+        feeDelegatedTransaction.appendFeePayerSignatures(tx.getFeePayerSignatures());
+
+        return feeDelegatedTransaction;
+    }
+
+    /**
+     * Getter function for walletAPI
+     * @return Wallet
+     */
+    public Wallet getWalletAPI() {
+        return walletAPI;
+    }
+
+    /**
+     * Setter function for walletAPI
+     * @param walletAPI The WalletAPI instance to use KAS Wallet API.
+     */
+    public void setWalletAPI(Wallet walletAPI) {
+        this.walletAPI = walletAPI;
+    }
+
+
+    private boolean isWeightedMultiSigType(AbstractTransaction transaction, String address) throws IOException {
+        AccountKey res = transaction.getKlaytnCall().getAccountKey(address).send();
+        if(res == null) {
+            return true;
+        }
+
+        AccountKey.AccountKeyData accountKeyData = res.getResult();
+        String type = accountKeyData.getType();
+
+        if(type.equals(AccountKeyWeightedMultiSig.getType())) {
+            return true;
+        }
+
+        if(type.equals(AccountKeyRoleBased.getType())) {
+            AccountKeyRoleBased roleBased = (AccountKeyRoleBased) accountKeyData.getAccountKey();
+
+            if(transaction instanceof AccountUpdate) {
+                if(roleBased.getRoleAccountUpdateKey() instanceof AccountKeyWeightedMultiSig) {
+                    return true;
+                }
+            } else {
+                if(roleBased.getRoleTransactionKey() instanceof  AccountKeyWeightedMultiSig) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private boolean isWeightedMultiSigType(AbstractFeeDelegatedTransaction fdTransaction, String address) throws IOException {
```

Comment:
What happen if someone want to sign fee delegated transaction as a sendeR? not as a fee payer?